drivers/SmartThings/matter-thermostat/src/init.lua

```diff
@@ -1161,6 +1166,9 @@ local function sequence_of_operation_handler(driver, device, ib, response)
   device:set_field(DISALLOWED_THERMOSTAT_MODES, disallowed_mode_operations)
   local event = capabilities.thermostatMode.supportedThermostatModes(supported_modes, {visibility = {displayed = false}})
   device:emit_event_for_endpoint(ib.endpoint_id, event)
+
+  -- do an extra SystemMode read in case of out of order attribute handling
+  device:send(clusters.Thermostat.attributes.SystemMode:read(device))
```

Comment:
Is an extra read neccessary? We have all the information we need at this point, right? So couldn't we just call into a helper function with the relevant information such as the supported modes and the optional modes seen? The same helper function could be used by the attribute handler as well, I'm just not sure why we require an additional device read